### PR TITLE
refactor(next): updated auth API status codes for improved error hand…

### PR DESCRIPTION
**Pull request by edtorba**
#4909

### CODE REVIEW

File: packages/payload/src/auth/operations/logout.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export const logoutOperation = async (incomingArgs: Arguments): Promise<boolean>
 
   if (!user) throw new APIError('No User', httpStatus.BAD_REQUEST)
   if (user.collection !== collectionConfig.slug)
-    throw new APIError('Incorrect collection', httpStatus.FORBIDDEN)
+    throw new APIError('Incorrect collection', httpStatus.INTERNAL_SERVER_ERROR)
```
Comment (JarrodMFlesch) [inline packages/payload/src/auth/operations/logout.ts:23]: I tend to think this should be `FORBIDDEN`. What do you think?

**Review by JarrodMFlesch — COMMENTED** [review]
Thanks for making these changes! See my comments below.

File: packages/payload/src/auth/operations/logout.ts (hunk 1 above)
Comment (edtorba) [reply packages/payload/src/auth/operations/logout.ts:23]: Hmm, might be wrong, but, I'm actually leaning towards using 404 now, as specifying the wrong collection technically means that we are attempting to communicate with something that does not exist.

Comment (edtorba) [reply packages/payload/src/auth/operations/logout.ts:23]: The use of 403 also feels somewhat incorrect, in my opinion. It sort of implies that the system understands the request and everything is correct but not valid for some reason, hence it refuses to process it. Whereas specifying a wrong collection is more like - "Oh well, but that does not exist..."

Comment (edtorba) [reply packages/payload/src/auth/operations/logout.ts:23]: And then the message could be changed to "Action not found" or something along those lines.

Comment (edtorba) [reply packages/payload/src/auth/operations/logout.ts:23]: Unless I am misunderstanding it 🤣

Comment (JarrodMFlesch) [reply packages/payload/src/auth/operations/logout.ts:23]: So the user is attempting to authenticate against the wrong auth collection.

Comment (edtorba) [reply packages/payload/src/auth/operations/logout.ts:23]: Okay, I'll revert it to 403 👍

**Pull request merged**